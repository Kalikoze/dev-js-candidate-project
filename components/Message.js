import React from 'react';
import PropTypes from 'prop-types';

const Message = ({ text }) => (
  <section className='message'>
    <p>{ text }</p>
    <style jsx>{`
      .message {
        background: rgba(255, 255, 255, 0.7);
        border-radius: .6em;
        font-family: 'Lato', sans-serif;
        font-size: 1.5em;
        height: auto;
        margin: 20px;
        padding: 25px;
        position: relative;
        width: 50%;
      }

      .message:after {
        border: 10px solid transparent;
        border-right-color: rgba(255, 255, 255, 0.7);
        border-bottom: 0;
        border-left: 0;
        content: '';
        height: 0;
        left: 0;
        margin-left: -10px;
        position: absolute;
        top: 70%;
        width: 0;
      }
    `}</style>
  </section>
);

Message.propTypes = {
  text: PropTypes.string.isRequired,
};

export default Message;
