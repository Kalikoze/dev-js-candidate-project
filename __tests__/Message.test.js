import React from 'react';
import Message from '../components/Message';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';

configure({ adapter: new Adapter() });

describe('Message component', () => {
  it('should render the Message correctly with the right classes and styles', () => {
    const tree = renderer.create(<Message isUser={false} text='Hello SpruceBot!' />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should have isUser and text as props', () => {
    const wrapper = mount(<Message isUser={true} text='Hello SpruceBot!' />);

    expect(wrapper.props()).toHaveProperty('isUser', true);
    expect(wrapper.props()).toHaveProperty('text', 'Hello SpruceBot!');
  });
});
