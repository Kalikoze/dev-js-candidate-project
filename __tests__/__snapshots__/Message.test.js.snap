// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component should render the Message correctly with the right classes and styles 1`] = `
<section
  className="jsx-2578047880 message watson"
>
  <p
    className="jsx-2578047880"
  >
    Hello SpruceBot!
  </p>
</section>
`;
